components/x-interaction/readme.md

```diff
@@ -108,3 +108,114 @@ export const BaseGreeting = ({greeting, actions}) => <div>
 
 export const Greeting = greetingActions(BaseGreeting);
 ```
+
+### Hydrating server-rendered markup
+
+When you have an `x-interaction` component rendered by the server, and you want to attach the client-side version of the component to handle the actions, rather than rendering the component manually (which might become unwieldy, especially if you have many components & instances on the page), you can have `x-interaction` manage it for you.
+
+There are two parts to this: serialising and hydrating.
+
+#### Serialising
+
+To ensure components are rendered with the same initial data on the client side, and keep track of all the instances that are rendered and their identifiers, you must create a serialiser on the server side. `x-interaction` exports a function `getInteractionSerialiser`. You *must* call this function *exactly once* at the start every HTTP request that responds with any number of `x-interaction` components.
```

Comment:
Not quite sure about the final sentence here, if this function returns a new instance then perhaps that could be mentioned?

> You must create a new instance of the serialiser for each request which responds with any `x-interaction` components.

Reply:
it's not exactly a new instance because it's pretty much a global singleton, but i want to pretend it's a new instance. i don't particularly like the api we currently have for this but i can't think of a way of not having it as a global (we can't rely on `context` or similar)